osbuild/solver/dnf.py

```diff
@@ -110,21 +115,27 @@ def __init__(self, request, persistdir, cache_dir, license_index_path=None):
         self.license_index_path = license_index_path
 
     @staticmethod
-    def _dnfrepo(desc, parent_conf=None):
+    def _dnfrepo(desc, parent_conf=None, root_dir=None):
         """Makes a dnf.repo.Repo out of a JSON repository description"""
 
         repo = dnf.repo.Repo(desc["id"], parent_conf)
+        config = libdnf.conf.ConfigParser
 
         if "name" in desc:
             repo.name = desc["name"]
 
+        def subs(basestr):
+            if root_dir:
+                return config.substitute(ucd(basestr), parent_conf.substitutions)
```

Comment:
`parent_conf` can be `None`. Either you need to check its value to prevent `AttributeError`, or change the `_dnfrepo()` definition to not set it to `None` by default.